.github/labeler.yml

```diff
@@ -0,0 +1,73 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+# Pull Request Labeler Github Action Configuration: https://github.com/marketplace/actions/labeler
+INFRA:
+  - .asf.yaml
+  - .gitattributes
+  - .gitignore
+  - baseline.gradle
+  - deploy.gradle
+  - jitpack.yml
+  - travis.yml
+  - .baseline/**/*
+  - .github/**/*
+  - dev/**/*
+BUILD:
+  - "**/*gradle*"
+  - versions.props
+DOCS:
+  - site/**/*
+  - "**/*CHANGELOG.md"
+  - "**/*README.md"
+EXAMPLES:
+  - examples/**/*
+COMMON:
+  - common/**/*
+API:
+  - api/**/*
+CORE:
+  - core/**/*
+PYTHON:
+  - python/**/*
+PARQUET:
+  - parquet/**/*
+ARROW:
+  - arrow/**/*
+ORC:
+  - orc/**/*
+HIVE:
+  - hive3/**/*
+  - hive-metastore/**/*
+  - hive-runtime/**/*
+DATA:
+  - data/**/*
+SPARK:
+  - spark-runtime/**/*
+  - spark3-runtime/**/*
+  - spark/**/*
+  - spark2/**/*
+  - spark3/**/*
+FLINK:
+  - flink-runtime/**/*
+  - flink/**/*
+MR:
+  - mr/**/*
+PIG:
+  - pig/**/*
+
```

Comment:
This whole file has been rather thoroughly tested in my personal fork, as outlined in the last few comments of the corresponding issue.

Reply:
It feels like there should be a way to say "All top level folders are modules and label accordingly" but If there isn't I think this is :+1:

Reply:
Yeah there's not unfortunately. In fact, their docs said if we wanted a top level folder named `foo`, its contents, and all of its subdirectories contents to be labeled, we should have to do the following two line S3 permissions inspired monstrosity (though I tested it and that's not the case):

```yaml
FOO:
  - foo/*
  - foo/**/*
```